README.md

```diff
@@ -1,145 +1,308 @@
+
+See the [Connecting Overview][connection-overview] page for more information on
+connecting to a Cloud SQL instance, or the [About the proxy][about-proxy] page
+for details on how the Cloud SQL proxy works.
+
+Note: The Proxy *cannot* provide a network path to a Cloud SQL instance if one
+is not already present (e.g., the proxy cannot access a VPC if it does not
+already have access).
+
+## Installation
+
+The proxy may be installed from the [releases page][releases].
+
+For alternative distributions, see below under [third party](#third-party).
+
+### Container Images
+
+As a convenience, there are containerized versions of the proxy available from
+the following Google Cloud Container Registry repositories:
+
+* `gcr.io/cloudsql-docker/gce-proxy`
+* `us.gcr.io/cloudsql-docker/gce-proxy`
+* `eu.gcr.io/cloudsql-docker/gce-proxy`
+* `asia.gcr.io/cloudsql-docker/gce-proxy`
+
+Each image is tagged with the associated proxy version. The following tags are
+currently supported:
+
+* `$VERSION` - default image (recommended)
+* `$VERSION-alpine` - uses [`alpine:3`](https://hub.docker.com/_/alpine)
+  as a base image (only supported from v1.17 up)
+* `$VERSION-buster` - uses [`debian:buster`](https://hub.docker.com/_/debian)
+  as a base image (only supported from v1.17 up)
+
+We recommend using the latest version of the proxy and updating the version
+regularly. However, we also recommend pinning to a specific tag and avoid the
+latest tag. Note: the tagged version is only that of the proxy. Changes in base
+images may break specific setups, even on non-major version increments. As such,
+it's a best practice to test changes before deployment, and use automated
+rollbacks to revert potential failures.
+
+### Install from Source
+
+To install from source, ensure you have the latest version of [Go
+installed](https://golang.org/doc/install).
+
+Then, simply run:
+
+```
+go get github.com/GoogleCloudPlatform/cloudsql-proxy/cmd/cloud_sql_proxy
+```
+
+The `cloud_sql_proxy` will be placed in `$GOPATH/bin` after `go get` completes.
+
+## Usage
+
+All the following invocations assume valid credentials are present in the
+environment. For more thorough documentation on the available flags,
+see the [CLI usage docs](docs/cli-usage.md). For more information on using the
+proxy as a library, see the [associated docs](docs/library-usage.md).
+
+The following examples all reference an `INSTANCE_CONNECTION_NAME`, which takes
+the form: `myproject:myregion:myinstance`. To find the
+`INSTANCE_CONNECTION_NAME`, run `gcloud sql instances describe <INSTANCE_NAME>`
+where `INSTANCE_NAME` is the name of the database instance.
+
+### TCP socket example
+
+``` bash
+# Starts the proxy listening on 127.0.0.1:5432
+cloud_sql_proxy -instances=<INSTANCE_CONNECTION_NAME>=tcp:5432
+```
+
+### Unix socket example
+
+``` bash
+# Must create a /cloudsql directory first
+cloud_sql_proxy -dir=/cloudsql -instances=<INSTANCE_CONNECTION_NAME>
+```
+
```

Comment:
nit: Add a line here that says something like "This will cause the proxy to mount a Unix domain socket a `/cloudsql/<INSTANCE_CONNECTION_NAME>`